### ISSUE TITLE
[TNS 3] Wrapped text in Button not properly centered (iOS)

### ISSUE THREAD

**Issue by toddanglin** [NativeScript/NativeScript#4266]
There seems to be a regression with the way wrapped text is centered in a NativeScript button on iOS.

Pre-TNS3, if you set `style="text-align:center;"` on a `Button` with `textWrap="true"`, the text would be both A) centered in the Button, and B) any wrapped text would be centered on the new wrapped line.

With TNS3 (confirmed in 3.0.1 on iOS), those same style settings will center the text in the Button, BUT wrapped text on new lines **will not** be centered. Instead, wrapped text is left aligned.

The problem should be easy to reproduce using any button with style settings that cause the text to wrap. For example:

```
<Button text="This is a test of a Button text wrapping" textWrap="true" style="text-align:center;width:100;" />
```
Does not apply to Android or `Label` widgets. Text is properly centered when wrapped.

**Comment by toddanglin** [comment]
Based on my limited mastery of the underlying native iOS properties, I *think* this issue can be fixed by making a small change in this file: https://github.com/NativeScript/NativeScript/blob/master/tns-core-modules/ui/button/button.ios.ts#L146

In this `textAlignmentProperty` implementation, you should be able to change this:
```typescript
switch (value) {
   case "left":
      this.nativeView.contentHorizontalAlignment = UIControlContentHorizontalAlignment.Left;
      break;
   case "initial":
   case "center":
      this.nativeView.contentHorizontalAlignment = UIControlContentHorizontalAlignment.Center;
      break;
   case "right":
      this.nativeView.contentHorizontalAlignment = UIControlContentHorizontalAlignment.Right;
      break;
}
```
To this:
```typescript
switch (value) {
   case "left":
      this.nativeView.titleLabel.textAlignment = NSTextAlignment.Left;
      break;
   case "initial":
   case "center":
      this.nativeView.titleLabel.textAlignment = NSTextAlignment.Center;
      break;
   case "right":
      this.nativeView.titleLabel.textAlignment = NSTextAlignment.Right;
      break;
}
```
I'd submit a PR directly, but not sure if there are reasons that using this approach was avoided in the first place. In my local tests, this does fix the text centering problem. Not sure if it creates other problems I'm missing.

**Comment by NickIliev** [comment]
@toddanglin to demonstrate what is the difference between the original implementation and your suggestions I am attaching two screenshots.

Original implementation
![screen shot 2017-05-26 at 1 45 28 pm](https://cloud.githubusercontent.com/assets/18008302/26491829/d4820eec-4219-11e7-96ba-baf129533b6d.png)

Todd's code
![screen shot 2017-05-26 at 1 44 25 pm](https://cloud.githubusercontent.com/assets/18008302/26491843/e0377cfe-4219-11e7-914d-f14db9b1355c.png)

**Comment by NickIliev** [comment]
@toddanglin we've  discussed this case and it seems that it would be great to use the wrapping as you have suggested. Let me now when you have a PR ready or let me know if you want for us to take care of it.

**Comment by toddanglin** [comment]
Okay. Let test with Icon + Text scenario to make sure this new approach doesn't cause any problems. If it doesn't, I'll go ahead and submit a quick PR for the file mentioned above with the new alignment code for iOS.

**Comment by toddanglin** [comment]
Okay. Confirmed that the new text alignment approach seems to work fine in iOS, including scenarios where the text includes unicode icons (see image below).

```xml
<Button text="This a test of a wrapping button [CENTER]" textWrap="true" style="text-align:center;width:100;margin-bottom:60;" />
<Button text="This a test of a wrapping button [RIGHT]" textWrap="true" style="text-align:right;width:100;margin-bottom:60;" />
<Button text="This a test of a wrapping button [LEFT]" textWrap="true" style="text-align:left;width:100;margin-bottom:60;" />
<Button text="&#xf8ff; This a test of a wrapping button [CENTER ICON]" textWrap="true" style="text-align:center;width:100;margin-bottom:60;" />
```
**Layout with new text alignment applied**
![image](https://cloud.githubusercontent.com/assets/647319/26507395/cdd2ea66-4215-11e7-9769-b2a550bb1fee.png)

**Layout with current un-patched behavior**
![image](https://cloud.githubusercontent.com/assets/647319/26507496/309bc5d2-4216-11e7-812e-6efa0ea00b59.png)

While testing, I did notice that the height of a button with wrapped text does not seem to be properly calculated in our layout system. When I change the `margin` from `60` to `20` in the above code snippet, it produces overlapping buttons like this:

![image](https://cloud.githubusercontent.com/assets/647319/26507546/72b347b0-4216-11e7-9661-91eeb325d43b.png)

This occurs both with and without my proposed PR, so it is probably a separate issue that needs to be investigated. Just flagging here for future follow-up.

Meanwhile, will prepare the official PR and submit shortly.

**Comment by toddanglin** [comment]
PR created: https://github.com/NativeScript/NativeScript/pull/4273

Do you want me to create a new issue for the "overlapping button" problem?

**Comment by NickIliev** [comment]
@toddanglin  I think it would be better to open new dedicated issue to the problem with the overlapping so we can track its fix correctly and associate the current one strictly with fixing the alignment,

**Comment by toddanglin** [comment]
Thanks @NickIliev.

I think we can close this issue now since the PR with the fix has been merged (#4273).

**Comment by SvetoslavTsenov** [comment]
Hey @toddanglin, the fix is already available in tns-core-modules@next and will also be released with tns-core-modules@3.1.0

**Comment by bearoutthere** [comment]
I'm on TNS 3.4.0 and am seeing the same problem with a label.  Wrapped text is ignoring horizontalAlignment="center" and left aligns.
